### PR TITLE
feat(data): distinguish between spot and blob

### PR DESCRIPTION
**Pull request by oetyng**
A `Blob` is at least 3072 bytes, as that is min self encryptable bytes.
A `Spot` is smaller than that, and will be encrypted with the client
encryption keys.
Both of these are type safe in that the construction of them will
validate their size.

### CODE REVIEW

File: src/client/client_api/blob_apis.rs [hunk 1]
```diff
@@ -136,12 +104,42 @@ impl Client {
...
+        encrypt_blob(blob.bytes(), owner.as_ref())
+    }
+
+    /// Encrypts a small piece of t(d)ata (spot) and returns the resulting address and the chunk.
+    /// Does not store anything to the network.
+    pub fn encrypt_spot(&self, spot: Spot, scope: Scope) -> Result<(SpotAddress, Chunk)> {
+        let encryption = encryption(scope, self.public_key());
+        let chunk = to_chunk(spot.bytes(), encryption.as_ref())?;
+        if chunk.value().len() >= self_encryption::MIN_ENCRYPTABLE_BYTES {
+            return Err(Error::Generic("You might need to pad the spot contents, as the encryption of it made it slightly too big".to_string()));
```
Comment (joshuef) [inline src/client/client_api/blob_apis.rs:120]: you might need to pay (aka add more bytes?) as the encryption made it _too big_?

I'm confused why more bytes gets us something smaller here. Maybe just the msg needs a tweak though

File: src/client/client_api/blob_apis.rs [hunk 2]
```diff
@@ -136,12 +104,42 @@ impl Client {
...
+    pub fn encrypt_blob(&self, blob: Blob, scope: Scope) -> Result<(BlobAddress, Vec<Chunk>)> {
         let owner = encryption(scope, self.public_key());
-        let (head_address, all_chunks) = get_data_chunks(data, owner.as_ref())?;
+        encrypt_blob(blob.bytes(), owner.as_ref())
+    }
+
+    /// Encrypts a small piece of t(d)ata (spot) and returns the resulting address and the chunk.
+    /// Does not store anything to the network.
+    pub fn encrypt_spot(&self, spot: Spot, scope: Scope) -> Result<(SpotAddress, Chunk)> {
+        let encryption = encryption(scope, self.public_key());
```
Comment (joshuef) [inline src/client/client_api/blob_apis.rs:117]: When I read this line, I think this is always going to encrypt.

I code w/ rust_analyzer, I can see it's an option. But skimming it's not always clear (it took me a bit to see that `None` here is how we _dont_ encrypt public scope stuff.

Anyway. I'm basically wondering if there's a name tweak here that can help make it clear (and some docs on the `encryption` func)

Perhaps it just being `optional_encryption` or something? Not really, sure but just wanted to highlight something that was tripping me up here

File: src/client/client_api/blob_apis.rs (hunk 1 above)
Comment (oetyng) [reply src/client/client_api/blob_apis.rs:120]: If encrypting it made it slightly too big, say it was `3071` and become `3072` bytes (idk if that can happen but I suppose so), then unless they pad it they won't be able to store it.
So if they pad it with a byte, they can store it as a `Blob`.

File: src/client/client_api/blob_apis.rs (hunk 2 above)
Comment (oetyng) [reply src/client/client_api/blob_apis.rs:117]: Ah, this was simply from copy passing `encrypt_blob`, which will in fact always (self)encrypt it.

But with `Spots` that is only if it's private, so another name definitely needed.

**Review by joshuef — APPROVED** [review]

**Pull request merged**